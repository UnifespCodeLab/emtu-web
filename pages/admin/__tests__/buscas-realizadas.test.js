import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils'
import Vuetify from 'vuetify'

import AdminPerformedSearches from '~/pages/admin/buscas-realizadas.vue'

const localVue = createLocalVue()
let vuetify
let wrapper

class ResizeObserver {
  observe () {}
  unobserve () {}
}

describe('Pages / Admin / AdminPerformedSearches', () => {
  beforeEach(() => {
    jest.mock('vue-apexcharts', () => jest.fn())
    window.ResizeObserver = ResizeObserver

    vuetify = new Vuetify()

    wrapper = mount(AdminPerformedSearches, {
      localVue,
      vuetify,
      data () {
        return {
          categories: ['1', '2', '3']
        }
      },
      stubs: {
        'router-link': RouterLinkStub
      }
    })
  })

  it('should render the page', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should render the correct elements', () => {
    const dialogs = wrapper.findAllComponents({ name: 'v-dialog' })
    expect(dialogs.length).toBe(2)

    const autocompletes = wrapper.findAllComponents({ name: 'v-autocomplete' })
    expect(autocompletes.length).toBe(3)

    const datePickers = wrapper.findAllComponents({ name: 'v-date-picker' })
    expect(datePickers.length).toBe(0)

    const searchButton = wrapper.findComponent({ name: 'v-btn' })
    expect(searchButton.props('color')).toBe('primary')
  })

  it('should render the correct start date', () => {
    const textField = wrapper.findAllComponents({ name: 'v-text-field' }).at(0)
    expect(textField.props('value')).toBe(null)
  })

  it('should render the correct end date', () => {
    const endDate = new Date(Date.now() - new Date().getTimezoneOffset() * 60000)
      .toISOString()
      .substr(0, 10)

    const textField = wrapper.findAllComponents({ name: 'v-text-field' }).at(1)
    expect(textField.props('value')).toBe(endDate)
  })

  it('should not render the chart container', () => {
    const container = wrapper.find('.line-ranking__graph-wrapper')
    expect(container.exists()).toBe(false)
  })

  describe('when search button is clicked', () => {
    beforeEach(() => {
      const searchButton = wrapper.findAllComponents({ name: 'v-btn' }).at(0)
      searchButton.trigger('click')
    })

    it('should render the chart container', () => {
      const container = wrapper.find('.line-ranking__graph-wrapper')
      expect(container.exists()).toBe(true)
    })

    it('should render correct chart', () => {
      const chart = wrapper.findComponent({ name: 'AdminChart' })
      expect(chart.props('title')).toBe('Ranking de linhas')
      expect(chart.props('type')).toBe('bar')
      expect(chart.props('categories')).toEqual(['1', '2', '3'])
    })
  })
})
