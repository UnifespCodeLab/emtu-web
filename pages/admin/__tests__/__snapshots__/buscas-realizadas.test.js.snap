// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages / Admin / AdminPerformedSearches should render the page 1`] = `
<div
  class="performed-searches"
>
  <div
    class="performed-searches__filter"
  >
    <div
      class="v-input performed-searches__input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--placeholder"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot"
        >
          <div
            class="v-text-field__slot"
          >
            <label
              class="v-label theme--light"
              for="input-3"
              style="left: 0px; position: absolute;"
            >
              Período de início
            </label>
            <input
              aria-expanded="false"
              aria-haspopup="true"
              id="input-3"
              role="button"
              type="text"
            />
          </div>
          <div
            class="v-input__append-inner"
          >
            <div />
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="v-dialog__container"
    >
      <!---->
    </div>
     
    <div
      class="v-dialog__container"
    >
      <!---->
    </div>
    <div
      class="v-input performed-searches__input v-input--is-label-active v-input--is-dirty v-input--is-readonly theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--placeholder"
    >
      <div
        class="v-input__control"
      >
        <div
          class="v-input__slot"
        >
          <div
            class="v-text-field__slot"
          >
            <input
              aria-expanded="false"
              aria-haspopup="true"
              id="input-7"
              readonly="readonly"
              role="button"
              type="text"
            />
          </div>
          <div
            class="v-input__append-inner"
          >
            <div
              class="v-input__icon v-input__icon--clear"
            >
              <button
                aria-label="clear icon"
                class="v-icon notranslate v-icon--link mdi mdi-close theme--light"
                type="button"
              />
            </div>
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="v-input performed-searches__input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select v-autocomplete"
    >
      <div
        class="v-input__control"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="list-11"
          class="v-input__slot"
          role="combobox"
        >
          <div
            class="v-select__slot"
          >
            <label
              class="v-label theme--light"
              for="input-11"
              style="left: 0px; position: absolute;"
            >
              Origem
            </label>
            <input
              autocomplete="off"
              id="input-11"
              type="text"
            />
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-menu-down theme--light"
                />
              </div>
            </div>
            <input
              type="hidden"
            />
          </div>
          <div
            class="v-menu"
          >
            <!---->
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="v-input performed-searches__input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select v-autocomplete"
    >
      <div
        class="v-input__control"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="list-16"
          class="v-input__slot"
          role="combobox"
        >
          <div
            class="v-select__slot"
          >
            <label
              class="v-label theme--light"
              for="input-16"
              style="left: 0px; position: absolute;"
            >
              Destino
            </label>
            <input
              autocomplete="off"
              id="input-16"
              type="text"
            />
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-menu-down theme--light"
                />
              </div>
            </div>
            <input
              type="hidden"
            />
          </div>
          <div
            class="v-menu"
          >
            <!---->
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="v-input performed-searches__input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select v-autocomplete"
    >
      <div
        class="v-input__control"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="list-21"
          class="v-input__slot"
          role="combobox"
        >
          <div
            class="v-select__slot"
          >
            <label
              class="v-label theme--light"
              for="input-21"
              style="left: 0px; position: absolute;"
            >
              Linha
            </label>
            <input
              autocomplete="off"
              id="input-21"
              type="text"
            />
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-menu-down theme--light"
                />
              </div>
            </div>
            <input
              type="hidden"
            />
          </div>
          <div
            class="v-menu"
          >
            <!---->
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </div>
     
    <button
      class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--large primary"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        
      BUSCAR
    
      </span>
    </button>
  </div>
   
  <!---->
</div>
`;
